### PR TITLE
Code Review for [3pt] Implement mission joining - Actions #10, [1pt] Implement mission, [1pt] Switch badges for Missions - Conditional components #6 leaving - Actions #9, 

### PR DESCRIPTION
**Pull request by amiraabouhadid**
## Description

This is a web app where users can book rockets and missions 

### General requirements
- [x] There are no linter errors.
- [x] Followed the **Github flow**.
- [x] The documentation is professional.
### HTML/CSS/JS requirements

- [x] We followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- [x] We followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).
### Project requirements

- Create a reducer and action dispatcher for the "Join Mission" button. The logic here is practically the same as with rockets - you need to pass the mission's ID to the corresponding action and update the missions' state with the selected mission having a new key/value - reserved: true.
- Redux: Write actions and reducers for canceling rockets/dragons and leaving missions
- Missions that the user has joined already should show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button (as per design).

Thank you for taking the time to review this pull request.

### CODE REVIEW

File: src/redux/missions/missions.js [hunk 1]
```diff
@@ -1,14 +1,65 @@
...
+const reducer = (state = [], action) => {
   const { type, payload } = action;
   switch (type) {
     case GETMISSIONS:
+      let missionsSaved;
+      if (localStorage.getItem('missions')) {
+        missionsSaved = JSON.parse(localStorage.getItem('missions'));
+        return [...missionsSaved];
+      }
+      localStorage.setItem('missions', JSON.stringify(payload));
```
Comment (jssol) [inline src/redux/missions/missions.js:17]: This is a good idea, but we should not have a side effect in the reducer.
Plus, it is strictly said that we should not persist user data in the project requirements.

> __Technical set up**__
>  - Set up the repository on GitHub and use Gitflow.
>  - Set up React and Redux.
>  - Set up testing libraries(React Testing Library and Jest).
>  - Do not preserve user data.

Kindly check it [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md) near the bottom of the page.

Comment (amiraabouhadid) [reply src/redux/missions/missions.js:17]: @jssol  alright thanks for pointing it out, will update accordingly

**Review by jssol — CHANGES_REQUESTED** [review]
#### Nice job implementing these features :tada:
However, there is an issue that needs to be addressed.

Please check my inline comments to learn more.
Feel free to tag me @jssol for further explanation.
Good luck!

File: src/redux/missions/missions.js (hunk 1 above)
Comment (jssol) [reply src/redux/missions/missions.js:17]: Great

Comment (jssol) [reply src/redux/missions/missions.js]: @amiraabouhadid, By the way, I'm still awaiting your review in this [PR](https://github.com/amiraabouhadid/space_travelers_hub/pull/28)

Comment (amiraabouhadid) [reply src/redux/missions/missions.js]: @jssol hey yeah I reviewed it already

Comment (amiraabouhadid) [reply src/redux/missions/missions.js]: @jssol I have implemented the changes you requested, please re-review and let m know if I need to make more changes.
Thanks, 
Amira

File: src/redux/missions/missions.js [hunk 2]
```diff
@@ -1,14 +1,57 @@
+/* eslint-disable no-case-declarations */
 import axios from 'axios';
 
 const GETMISSIONS = 'app/missions/MISSIONS_RETRIEVED';
+const JOINMISSION = 'app/missions/MISSION_JOINED';
+const LEAVEMISSION = 'app/missions/MISSION_LEFT';
```
Comment (jssol) [inline src/redux/missions/missions.js:6]: Please format the action types according to the ducks pattern.
Action types var name should match the actual action name.

File: src/redux/missions/missions.js [hunk 3]
```diff
@@ -1,14 +1,57 @@
...
     default:
-      return missions;
+      return state;
+  }
+};
+export const leaveMission = (id) => (dispatch) => {
+  try {
+    dispatch({
+      type: LEAVEMISSION,
+      payload: id,
```
Comment (jssol) [inline src/redux/missions/missions.js:39]: Kindly refactor to using action creators according to the ducks pattern.

File: src/redux/missions/missions.js [hunk 4]
```diff
@@ -1,14 +1,57 @@
...
+    return Promise.resolve(id);
+  } catch (err) {
+    return Promise.reject(err);
+  }
+};
+export const joinMission = (id) => (dispatch) => {
+  try {
+    dispatch({
+      type: JOINMISSION,
+      payload: id,
```
Comment (jssol) [inline src/redux/missions/missions.js:50]: Kindly refactor to using action creators according to the ducks pattern.

**Review by jssol — CHANGES_REQUESTED** [review]
Great job implementing the previously required changes.
There are some issues I didn't notice at first but kindly check the inline comments to fix them quickly.

File: src/redux/missions/missions.js (hunk 2 above)
Comment (amiraabouhadid) [reply src/redux/missions/missions.js]: Done, please check

File: src/redux/missions/missions.js (hunk 4 above)
Comment (amiraabouhadid) [reply src/redux/missions/missions.js]: Done, please check

File: src/redux/missions/missions.js (hunk 3 above)
Comment (amiraabouhadid) [reply src/redux/missions/missions.js]: Done, please check

**Review by jssol — APPROVED** [review]
Nice job implementing the changes :100:
I am awaiting your review it the other PR.

**Pull request merged**